app/src/main/java/com/brandon/forzenbook/repository/ForzenRepositoryImpl.kt

```diff
@@ -7,13 +7,29 @@ import com.brandon.forzenbook.data.toForzenEntity
 import com.brandon.forzenbook.network.ForzenApiService
 import com.brandon.forzenbook.network.LoginRequest
 import com.brandon.forzenbook.network.toCreateAccountRequest
+import com.brandon.forzenbook.viewmodels.CreateUserOutcome
 
 class ForzenRepositoryImpl(
     private val forzenApiService: ForzenApiService,
     private val forzenDao: ForzenDao,
 ) : ForzenRepository {
 
-    override suspend fun getToken(userName: String, code: String): LoginToken? {
+    override suspend fun getCode(email: String): Boolean {
```

Comment:
Shouldn't THIS be what throws your custom exceptions? This boolean returning isn't helping. The repo should handle data. Make this throw and the use case catches it

Reply:
am now throwing here, additionally, adjusted the repo to simply store the token since the view doesn't really care about it directly